src/App.tsx

```diff
@@ -1,14 +1,55 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Status } from './types/Status';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+export const filteredQueryInput = (
+  comparedTodos: Todo[],
+  query: string,
+): Todo[] => {
+  const visibleTodos = [...comparedTodos];
+
+  if (query) {
+    return visibleTodos.filter(todo =>
+      todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
+  }
+
+  return visibleTodos;
+};
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+
+  const [eyeOnClick, setEyeOnClick] = useState(false);
```

Comment:
I have no idea what is the reason for this state, and why it has so a funny name. 
You can render the modal only when you have selected todo.